### PR TITLE
fix(kit): allow `src` for `addPluginTemplate`

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Previously, we actually errored if `src` was provided, e.g. https://github.com/nuxt/framework/blob/e674d0f60d52e26122ce54f6c82a5723f8fc8e23/packages/bridge/src/capi.ts#L26. This PR makes it optional.

### CODE REVIEW

File: packages/schema/src/types/nuxt.ts [hunk 1]
```diff
@@ -59,5 +59,5 @@ export interface NuxtApp {
   templates: NuxtTemplate[]
 }
 
-type _TemplatePlugin = NuxtPlugin & NuxtTemplate
+type _TemplatePlugin = Partial<NuxtPlugin> & NuxtTemplate
```
Comment (pi0) [inline packages/schema/src/types/nuxt.ts:62]: Does it produce proper jsdocs? `src` definition from Plugin is different from template

Comment (danielroe) [reply packages/schema/src/types/nuxt.ts:62]: Previously:

```ts
/**
 * Adds a template and registers as a nuxt plugin.
 */
declare function addPluginTemplate(plugin: Omit<NuxtPluginTemplate, 'src'> | string, opts?: AddPluginOptions): NuxtPluginTemplate;
```

Now:

```ts
/**
 * Adds a template and registers as a nuxt plugin.
 */
declare function addPluginTemplate(plugin: NuxtPluginTemplate | string, opts?: AddPluginOptions): NuxtPlugin;
```

Comment (danielroe) [reply packages/schema/src/types/nuxt.ts:62]: CleanShot 2022-03-08 at 16 30 36

Comment (pi0) [inline packages/schema/src/types/nuxt.ts:62]:
```suggestion
type _TemplatePlugin = Omit<NuxtPlugin, 'src'> & NuxtTemplate
```

**Pull request merged**